lib/json-parser.js

```diff
@@ -0,0 +1,250 @@
+  label: /unused type parameter/,
+  message: /type parameter `.+` is unused/
+}, {
+  // E0101
+  label: /cannot resolve type of expression/,
+  message: /cannot determine a type for this expression: unconstrained type/
+}, {
+  // E0102
+  label: /cannot resolve type of variable/,
+  message: /cannot determine a type for this local variable: unconstrained type/
+}, {
+  // E0106
+  label: /expected lifetime parameter/,
+  message: /missing lifetime specifier/
+}, {
+  // E0107
+  label: /(un)?expected (\d+ )?lifetime parameter[s]?/,
+  message: /wrong number of lifetime parameters: expected \d+, found \d+/
+}, {
+  // E0109
+  label: /type parameter not allowed/,
+  message: /type parameters are not allowed on this type/
+}, {
+  // E0110
+  label: /lifetime parameter not allowed/,
+  message: /lifetime parameters are not allowed on this type/
+}, {
+  // E0116
+  label: /impl for type defined outside of crate/,
+  message: /cannot define inherent `.+` for a type outside of the crate where the type is defined/
+}, {
+  // E0117
+  label: /impl doesn't use types inside crate/,
+  message: /only traits defined in the current crate can be implemented for arbitrary types/
+}, {
+  // E0119
+  label: /conflicting implementation for `.+`/,
+  message: /conflicting implementations of trait `.+` for type `.+`/
+}, {
+  // E0120
+  label: /implementing Drop requires a struct/,
+  message: /the Drop trait may only be implemented on structures/
+}, {
+  // E0121
+  label: /not allowed in type signatures/,
+  message: /the type placeholder `_` is not allowed within types on item signatures/
+}, {
+  // E0124
+  label: /field already declared/,
+  message: /field `.+` is already declared/
+}, {
+  // E0368
+  label: /cannot use `[<>+&|^\-]?=` on type `.+`/,
+  message: /binary assignment operation `[<>+&|^\-]?=` cannot be applied to type `.+`/
+}, {
+  // E0387
+  label: /cannot borrow mutably/,
+  message: /cannot borrow immutable local variable `.+` as mutable/
+}];
+
+// Copies a location from the given span to a linter message
+function copySpanLocation(span, msg) {
+  msg.file = span.file_name;
+  msg.line = span.line_start;
+  msg.line_end = span.line_end;
+  msg.col = span.column_start;
+  msg.col_end = span.column_end;
+}
+
+// Checks if the location of the given span is the same as the location
+// of the given message
+function compareLocations(span, msg) {
+  return span.file_name === msg.file
+    && span.line_start === msg.line
+    && span.line_end === msg.line_end
+    && span.column_start === msg.col
+    && span.column_end === msg.col_end;
+}
+
+// Appends spans's lable to the main message. It only adds the lable if:
+// - the main message doesn't contain exactly the same phrase
+// - the main message doesn't contain the same information but uses different wording
+function appendSpanLabel(span, msg) {
+  if (!span.label || msg.message.indexOf(span.label) >= 0) {
+    return;
+  }
+  for (let idx = 0; idx < redundantLabels.length; idx++) {
```

Comment:
In JS, `for (var l in redundantLabels)` iterates through indices, not through the array elements. At first glance, it looked counterintuitive to me, so I decided to leave the explicit iteration through indices.

`redundantLabels.forEach()` also doesn't work well here because it becomes too wordy when you need to return from the main function within a `forEach`-callback.